code_analysis.py

```
@@ -107,33 +124,200 @@ def __init__(self, project_path: str):
+        Args:
+            text: Text to embed
+            
+        Returns:
+            Numpy array containing the embedding
+        """
+        if not self.embedding_model:
+            return np.zeros((1, EMBEDDING_DIMENSION), dtype='float32')
+            
+        return self.embedding_model.encode([text])
+        
+    def add_document(self, content: str, metadata: Dict[str, Any] = None) -> int:
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)